core/src/test/scala/unit/kafka/admin/ConsumerGroupServiceTest.scala

```diff
@@ -62,6 +62,92 @@ class ConsumerGroupServiceTest {
     verify(admin, times(1)).listOffsets(offsetsArgMatcher, any())
   }
 
+  @Test
+  def testAdminRequestsForDescribeNegativeOffsets(): Unit = {
+    val args = Array("--bootstrap-server", "localhost:9092", "--group", group, "--describe", "--offsets")
+    val groupService = consumerGroupService(args)
+
+    val testTopicPartition0 = new TopicPartition("testTopic1", 0);
+    val testTopicPartition1 = new TopicPartition("testTopic1", 1);
+    val testTopicPartition2 = new TopicPartition("testTopic1", 2);
+    val testTopicPartition3 = new TopicPartition("testTopic2", 0);
+    val testTopicPartition4 = new TopicPartition("testTopic2", 1);
+    val testTopicPartition5 = new TopicPartition("testTopic2", 2);
+
+    val offsets = Map(
+      //testTopicPartition0 -> there is no offset information for an asssigned topic partition
+      testTopicPartition1 -> new OffsetAndMetadata(100), // regular information for a assigned partition
+      testTopicPartition2 -> null, //there is a null value for an asssigned topic partition
+      // testTopicPartition3 ->  there is no offset information for an unasssigned topic partition
+      testTopicPartition4 -> new OffsetAndMetadata(100), // regular information for a unassigned partition
+      testTopicPartition5 -> null, //there is a null value for an unasssigned topic partition
+    ).asJava
+
+    val resultInfo = new ListOffsetsResult.ListOffsetsResultInfo(100, System.currentTimeMillis, Optional.of(1))
+    val endOffsets = Map(
+      testTopicPartition0 -> KafkaFuture.completedFuture(resultInfo),
+      testTopicPartition1 -> KafkaFuture.completedFuture(resultInfo),
+      testTopicPartition2 -> KafkaFuture.completedFuture(resultInfo),
+      testTopicPartition3 -> KafkaFuture.completedFuture(resultInfo),
+      testTopicPartition4 -> KafkaFuture.completedFuture(resultInfo),
+      testTopicPartition5 -> KafkaFuture.completedFuture(resultInfo),
+    )
+    val assignedTopicPartitions = Set(testTopicPartition0, testTopicPartition1, testTopicPartition2 )
+    val unassignedTopicPartitions = offsets.asScala.filterNot { case (tp, _) => assignedTopicPartitions.contains(tp) }.toMap.keySet
```

Comment:
So here, we have `testTopicPartition4` and `testTopicPartition5` because `testTopicPartition3` is not in `offsets`. Is it expected to not have `testTopicPartition3`? Personally, I would rather prefer to list the partition explicitly here. It makes the test easier to read in my opinion.

Reply:
You are absolutely right, miss testTopicPartition3. Going to set it explicitly 👍